go-controller/pkg/clustermanager/network_cluster_controller.go

```diff
@@ -172,6 +224,20 @@ func (h *networkClusterControllerEventHandler) UpdateResource(oldObj, newObj int
 	var err error
 
 	switch h.objType {
+	case factory.PodType:
+		old, ok := oldObj.(*corev1.Pod)
+		if !ok {
+			return fmt.Errorf("could not cast %T old object to *corev1.Pod", oldObj)
+		}
+		new, ok := newObj.(*corev1.Pod)
+		if !ok {
+			return fmt.Errorf("could not cast %T new object to *corev1.Pod", newObj)
+		}
+		err := h.ncc.podIPAllocator.Reconcile(new, old)
```

Comment:
aren't the old / new pod arguments switched ? The function's signature is:
```golang
func (a *PodIPAllocator) Reconcile(old, new *corev1.Pod) error {
...
```
Src: https://github.com/ovn-org/ovn-kubernetes/pull/3620/files#diff-d0111b0bbab46d1b09ce2dcb9bf69046b81311a60da03baad0fce16c0c693d2dR74

Reply:
:facepalm: